torch/__init__.py

```diff
@@ -270,22 +270,30 @@ def set_default_tensor_type(t):
 
 
 def set_default_dtype(d):
-    r"""Sets the default floating point dtype to :attr:`d`. This type will be
-    used as default floating point type for type inference in
-    :func:`torch.tensor`.
+    r"""Sets the default floating point dtype to :attr:`d`.
+    This dtype is:
+    1. The inferred dtype for python floats in :func:`torch.tensor`.
+    2. Used to determine the default complex dtype. The default complex dtype is set to
+       ``torch.complex128`` if default floating point tensor type is ``torch.float64``,
+       otherwise it's set to ``torch.complex64``
```

Comment:
I'm curious: why did you word it this way instead of saying something about the corresponding float dtype?

Reply:
The only reason why I didn't it frame it that way is because we don't have a complex dtype corresponding to `torch.bfloat16`, `torch.float16` (technically we do but the support for ComplexHalf is almost non-existing)